### ISSUE TITLE
Optimise `attestation_pk` to not re-construct public keys so many times

### ISSUE THREAD

**Issue by thomaseizinger** [comit-network/maia#24]
Here is a flamegraph for a contract setup with the taker of `itchysats` at the time of writing this issue:

[flamegraph.zip](https://github.com/comit-network/maia/files/8347677/flamegraph.zip)

From that flamegraph, we can see that we spend a whopping 12% of our CPU time in the `attestation_pk` function. 2/3 of that (8% in total) are spent with parsing public keys.

I made a dirty patch of [maia](https://github.com/comit-network/maia/commit/d954ca5f37da329b9b4f849caa58413d361e379a) and [itchysats](https://github.com/itchysats/itchysats/commit/c3f51e949689a49eb2a71f50018f215520f8f37a) where I replaced all usages of `schnorrsig::Publickey` with `secp256k1_zkp::Publickey`. Here is the flamegraph of that:

[flamegraph.zip](https://github.com/comit-network/maia/files/8347694/flamegraph.zip)

In that flamegraph, `attestation_pk` takes only 4.5% of the entire CPU time but all of that is spent in combining keys (which is that actual job of that function).

In a nutshell, this means we can easily knock off 8% of CPU cycles for each contract-setup / rollover. From skimming the logs, the gain is about 2 seconds in execution time: From 11s to 9s between the logs `Exchanged setup parameters` and `Contract setup was completed`.

I didn't submit the patches because:

a) I think using x-only is more correct because that is what olivia is returning
b) Unfortunately, it is unlikely that `secp256k1` will support the required arithmetic to not convert the keys: https://github.com/rust-bitcoin/rust-secp256k1/issues/405
c) The patches are hacky

**Comment by thomaseizinger** [comment]
cc @luckysori @klochowicz You might be interested in these findings.
